src/QMCWaveFunctions/RotatedSPOs.cpp

```diff
@@ -626,6 +631,69 @@ void RotatedSPOs::log_antisym_matrix(const ValueMatrix& mat, ValueMatrix& output
     }
 }
 
+void RotatedSPOs::log_antisym_matrix(const Matrix<std::complex<RealType>>& mat, Matrix<std::complex<RealType>>& output)
+{
+  const int n = mat.rows();
+
+  //From here on out, all temporary arrays will be column major to interface with LAPACK cleanly.
+  //
+  //Temporary arrays for the eigenvalue decomposition.
+  std::vector<std::complex<RealType>> mat_h(n * n, 0);
+  std::vector<std::complex<RealType>> eval(n, 0);      //eigenvalues
+  std::vector<std::complex<RealType>> mat_l(n * n, 0); //left eigenvectors.
+  //complex work array.  Recommended to be at least 2*n according to LAPACK documentation.
+  std::vector<std::complex<RealType>> work(2 * n, 0);
+  std::vector<RealType> rwork(2 * n, 0);
+
+  std::vector<std::complex<RealType>> mat_cd(n * n, 0);
+  std::vector<std::complex<RealType>> mat_cl(n * n, 0);
+  std::vector<std::complex<RealType>> mat_ch(n * n, 0);
+
+  //Convert row major mat to column major mat_h.
+  for (int i = 0; i < n; ++i)
+    for (int j = 0; j < n; ++j)
+      mat_h[i + n * j] = mat[i][j];
+
+  // diagonalize the matrix
+  char JOBL('V');
+  char JOBR('N');
+  int N(n);
+  int LDA(n);
+  int LWORK(4 * n);
+  int info = 0;
+  //Generalized eigenvalue decomposition for A*x=lambda x.
+  LAPACK::geev(&JOBL, &JOBR, &N, &mat_h.at(0), &LDA, &eval.at(0), &mat_l.at(0), &LDA, nullptr, &LDA, &work.at(0),
+               &LWORK, &rwork.at(0), &info);
+  if (info != 0)
+  {
+    std::ostringstream msg;
+    msg << "zgeev failed with info = " << info << " in RotatedSPOs::log_antisym_matrix";
+    throw std::runtime_error(msg.str());
+  }
+
+  // iterate through diagonal matrix, take log
+  for (int i = 0; i < n; ++i)
+  {
+    for (int j = 0; j < n; ++j)
+    {
+      auto tmp          = (i == j) ? std::log(eval[i]) : std::complex<RealType>(0.0, 0.0);
+      mat_cd[i + j * n] = tmp;
+    }
+  }
+
+  RealType one(1.0);
```

Comment:
Since RealType is real guaranteed here why not just use the literals?

Still it is less awful than
`RealType cone(1.0);`
Which combines this idiom usually only used with types that can be real or complex with a variable name that I once assumed meant `complex 1` in QMCPACK

Reply:
Done